pkg/apis/options/providers.go

```diff
@@ -0,0 +1,172 @@
+	RedeemURL string `json:"redeemURL,omitempty"`
+	// ProfileURL is the profile access endpoint
+	ProfileURL string `json:"profileURL,omitempty"`
+	// ProtectedResource is the resource that is protected (Azure AD only)
+	ProtectedResource string `json:"resource,omitempty"`
+	// ValidateURL is the access token validation endpoint
+	ValidateURL string `json:"validateURL,omitempty"`
+	// Scope is the OAuth scope specification
+	Scope string `json:"scope,omitempty"`
+	// Prompt is OIDC prompt
+	Prompt string `json:"prompt,omitempty"`
+	// ApprovalPrompt is the OAuth approval_prompt
+	// default is set to 'force'
+	ApprovalPrompt string `json:"approvalPrompt,omitempty"`
+	// AllowedGroups is a list of restrict logins to members of this group
+	AllowedGroups []string `json:"allowedGroups,omitempty"`
+
+	// AcrValues is a string of acr values
+	AcrValues string `json:"acrValues,omitempty"`
+}
+
+type KeycloakOptions struct {
+	//KeycloakGroup enables to restrict login to members of indicated group
+	KeycloakGroup string `json:"keycloakGroup,omitempty"`
+}
+
+type AzureOptions struct {
+	// AzureTenant directs to a tenant-specific or common (tenant-independent) endpoint
+	// Default value is 'commmon'
+	AzureTenant string `json:"azureTenant,omitempty"`
+}
+
+type BitbucketOptions struct {
+	// BitbucketTeam sets restrict logins to members of this team
+	BitbucketTeam string `json:"bitbucketTeam,omitempty"`
+	// BitbucketRepository sets restrict logins to user with access to this repository
+	BitbucketRepository string `json:"bitbucketRepository,omitempty"`
+}
+
+type GitHubOptions struct {
+	// GitHubOrg sets restrict logins to members of this organisation
+	GitHubOrg string `json:"githubOrg,omitempty"`
+	// GitHubTeam sets restrict logins to members of this team
+	GitHubTeam string `json:"githubTeam,omitempty"`
+	// GitHubRepo sets restrict logins to collaborators of this repository
+	GitHubRepo string `json:"githubRepo,omitempty"`
+	// GitHubToken is the token to use when verifying repository collaborators
+	// it must have push access to the repository
+	GitHubToken string `json:"githubToken,omitempty"`
+	// GitHubUsers allows users with these usernames to login
+	// even if they do not belong to the specified org and team or collaborators
+	GitHubUsers []string `json:"githubUsers,omitempty"`
+}
+
+type GitLabOptions struct {
+	// GitLabGroup sets restrict logins to members of this group
+	GitLabGroup []string `json:"gitlabGroups,omitempty"`
+}
+
+type GoogleOptions struct {
+	// GoogleGroups sets restrict logins to members of this google group
+	GoogleGroups []string `json:"googleGroup,omitempty"`
+	// GoogleAdminEmail is the google admin to impersonate for api calls
+	GoogleAdminEmail string `json:"googleAdminEmail,omitempty"`
+	// GoogleServiceAccountJSON is the path to the service account json credentials
+	GoogleServiceAccountJSON string `json:"googleServiceAccountJson,omitempty"`
+}
+
+type OIDCOptions struct {
```

Comment:
Take a look at this issue: https://github.com/oauth2-proxy/oauth2-proxy/issues/956

Eventually, I want to move the Azure, Google, Keycloak, LoginGov & Gitlab providers to be completely based on the OIDC provider.

Does this change how you want to structure this section?

Also, existing users use some of these (`OIDCIssuerURL` in particular) to extract various URLs for other providers and get IDTokenVerifier support (I believe `keycloak` and `gitlab` users for sure do this). Will this organization affect that mixed usage at all?

Gitlab in particular is currently an OIDC provider (with the implementation unfortunately duplicated from the `OIDCProvider`) and needs these flags as well.

Reply:
Well my thought was that every provider will have its own set of options and as set of OIDC options so that if any OIDC related option will be set, they will propagate to the provider that is being set. But I might be missing something here, what did you have in mind for the structure?

Reply:
It sounds like we are good, as long as these options aren't solely tagged to the OIDC provider and are spread and consumed throughout the other providers that use them, we are good.